File: WebApplication/src/nomenclatures/dtos/nomenclature-hierarchy.dto.ts
import { Level } from "src/shared/enums/level.enum";
import { NomenclatureDto } from "./nomenclature.dto";

export class NomenclatureHierarchyDto<T extends NomenclatureHierarchyDto<T>> extends NomenclatureDto {
    rootId: number;
    root: T;
    parentId: number;
    parent: T;
    level: Level;
}

File: Sc.Models/Dtos/Base/Nomenclatures/NomenclatureHierarchyDto.cs
using Sc.Models.Enums.Common;

namespace Sc.Models.Dtos.Base.Nomenclatures
{
    public abstract class NomenclatureHierarchyDto<T> : NomenclatureDto
        where T : NomenclatureHierarchyDto<T>
    {
        public int? RootId { get; set; }
        public T Root { get; set; }
        public int? ParentId { get; set; }
        public T Parent { get; set; }
        public Level Level { get; set; }
    }
}
